overlay.go

```diff
@@ -331,3 +335,321 @@ func isZero(v reflect.Value) bool {
 		panic(fmt.Sprintf("unexpected value of type %s passed to isZero", v.Kind().String()))
 	}
 }
+
+// ReadAndMergeBundleData reads N bundle data sources, composes their contents
+// together and returns the result. The first bundle data source is treated as
+// a base bundle while subsequent bundle data sources are treated as overlays
+// which are sequentially merged onto the base bundle.
+//
+// Before returning the merged bundle, ReadAndMergeBundleData will also attempt
+// to resolve any include directives present in the machine annotations,
+// application options and annotations by invoking the provided includeResolver
+// with the path to the file required for each include.
+//
+// When merging an overlay into a base bundle the following rules apply for the
+// BundleData struct fields:
+// - if an overlay specifies a bundle-level series, it overrides the base bundle
+//   series.
+// - overlay-defined relations are appended to the base bundle relations
+// - overlay-defined machines overwrite the baase bundle machines.
+// - if an overlay defines an application that is not present in the base bundle,
+//   it will get appended to the application list.
+// - if an overlay defines an empty/nil application, it will be removed from the
```

Comment:
if an overlay defines an application with no contents ?
Otherwise it sounds like the app is nil, not the contents of the app declaration. I'm not sure if it is a lot better.

Reply:
The proposed implementation can work with both ways because the removal predicate is:
`defined-in-overlay-field-map && isZeroOrNil(val)`.

Reply:
My point is more just phrasing on the doc. "what is a nil application". Maybe others wouldn't find the wording confusing.